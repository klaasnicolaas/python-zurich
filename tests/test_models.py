"""Test the models."""

from __future__ import annotations

from typing import TYPE_CHECKING

from aresponses import ResponsesMockServer
from syrupy.assertion import SnapshotAssertion

from . import load_fixtures

if TYPE_CHECKING:
    from zurich import DisabledParking, ODPZurich


async def test_disabled_parkings(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    odp_zurich_client: ODPZurich,
) -> None:
    """Test disabled parking spaces function."""
    aresponses.add(
        "www.ogd.stadt-zuerich.ch",
        "/wfs/geoportal/Behindertenparkplaetze",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/vnd.geo+json"},
            text=load_fixtures("disabled_parkings.json"),
        ),
    )
    spaces: list[DisabledParking] = await odp_zurich_client.disabled_parkings()
    assert spaces == snapshot
