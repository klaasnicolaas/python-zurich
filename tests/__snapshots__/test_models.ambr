# serializer version: 1
# name: test_disabled_parkings
  list([
    DisabledParking(spot_id=206880, address='Ueberlandstrasse 460', longitude=8.594658, latitude=47.404414),
    DisabledParking(spot_id=206881, address='Ueberlandstrasse 460', longitude=8.594597, latitude=47.404434),
    DisabledParking(spot_id=206914, address='Querstrasse 16', longitude=8.544245, latitude=47.409697),
    DisabledParking(spot_id=206915, address='Querstrasse 16', longitude=8.54428, latitude=47.409706),
    DisabledParking(spot_id=206916, address='Greifenseestrasse 10a', longitude=8.553921, latitude=47.40894),
    DisabledParking(spot_id=206917, address='Zwyssigstrasse 2', longitude=8.485819, latitude=47.384643),
    DisabledParking(spot_id=206918, address='Bristenstrasse 17', longitude=8.488286, latitude=47.38997),
    DisabledParking(spot_id=206919, address='MÃ¼hlebachstrasse 137', longitude=8.557459, latitude=47.358313),
    DisabledParking(spot_id=206920, address='Wilfriedstrasse 5', longitude=8.554313, latitude=47.370063),
    DisabledParking(spot_id=206921, address='Friesenberghalde 8', longitude=8.501746, latitude=47.361433),
    DisabledParking(spot_id=206922, address='Friesenberghalde 3', longitude=8.501921, latitude=47.36132),
  ])
# ---
